include/index.h

```diff
@@ -14,7 +13,96 @@
 #include <seqan3/alphabet/nucleotide/all.hpp>
 
 
+
+namespace index_structure
+{
+
+    using ibf = seqan3::interleaved_bloom_filter<seqan3::data_layout::uncompressed>;
+    using ibf_compressed = seqan3::interleaved_bloom_filter<seqan3::data_layout::compressed>;
+
+    template <typename return_t, typename input_t>
+    concept compressible_from = (std::same_as<return_t, ibf_compressed> && std::same_as<input_t, ibf>);
+
+} // namespace index_structure
+
+class IndexStructure
+{
+
+private:
+    uint8_t bin_count_;
+    uint32_t bin_size_{};
+    uint8_t hash_count_{};
+    seqan3::interleaved_bloom_filter<seqan3::data_layout::uncompressed> ibf_;
+
+public:
+
+    uint8_t k_;
+
+    IndexStructure() = default;
+
+    explicit IndexStructure(uint8_t k,
+                            uint8_t bc,
+                            uint32_t bs,
+                            uint8_t hc) :
+            k_{k},
+            bin_count_{bc},
+            bin_size_{bs},
+            hash_count_{hc},
+            ibf_{seqan3::bin_count{bin_count_},
+                 seqan3::bin_size{bin_size_},
+                 seqan3::hash_function_count{hash_count_}}
+    {
+        //static_assert(data_layout_mode == seqan3::data_layout::uncompressed);
+    }
+
+    uint8_t getBinCount()
```

Comment:
```suggestion
    uint8_t getBinCount() const
```

The `const` guarantees you that you don't accidentally write `bin_conut_ = 5` in this function.
You can do something similar for `getBinSize()` and `getHashCount()`